dartsim/src/FreeGroupFeatures.cc

```diff
@@ -47,6 +48,29 @@ Identity FreeGroupFeatures::FindFreeGroupForModel(
     }
   }
 
+  for (const auto &nestedModel : modelInfo->nestedModels)
+  {
+    // Check if any of the nested models have valid free groups by recursively
+    // calling FindFreeGroupForModel on all nested models that either have
+    // BodyNodes or nested models. A nested model that doesn't have any
+    // BodyNodes or nested models should not exist per the SDFormat spec.
+    // However, BodyNodes may be moved out of a skeleton when creating joints
+    // leaving the skeleton without any BodyNodes. In this case, we skip the
+    // skeleton because the free group of the BodyNodes it used to contain will
+    // be determined when calling FindFreeGroupForModel for the model that now
+    // contains them after the move.
```

Comment:
the wording could probably be improved here; I had to reread it a couple times as it's a bit confusing. A suggestion:

"Check that each nested model with BodyNodes or nested models has valid free groups by recursively calling FindFreeGroupForModel. Nested models without BodyNodes or their own nested models are disallowed by the SDFormat spec, yet may occur if all BodyNodes are moved out of a skeleton when creating joints. In this case, skip such a skeleton instead of returning an error."

Reply:
Thanks for the suggestion. https://github.com/ignitionrobotics/ign-physics/pull/231/commits/74b3f6d75d66eef7fb9af2caf5697e1629cb5046